packages/cli/src/cli-entry-point.spec.ts

```diff
@@ -5,29 +5,48 @@ import 'reflect-metadata';
 import { Container } from 'inversify';
 import { IMock, Mock } from 'typemoq';
 import { CliEntryPoint } from './cli-entry-point';
-import { CommandRunner } from './command-runner';
+import { FileCommandRunner } from './runner/file-command-runner';
+import { URLCommandRunner } from './runner/url-command-runner';
 import { ScanArguments } from './scanner/scan-arguments';
 
 describe(CliEntryPoint, () => {
     let testSubject: CliEntryPoint;
     let containerMock: IMock<Container>;
-    let commandRunnerMock: IMock<CommandRunner>;
+    let urlCommandRunnerMock: IMock<URLCommandRunner>;
+    let fileCommandRunnerMock: IMock<FileCommandRunner>;
 
     beforeEach(() => {
         containerMock = Mock.ofType(Container);
-        commandRunnerMock = Mock.ofType(CommandRunner);
+        urlCommandRunnerMock = Mock.ofType(URLCommandRunner);
+        fileCommandRunnerMock = Mock.ofType(FileCommandRunner);
 
         testSubject = new CliEntryPoint(containerMock.object);
     });
 
     describe('runScan', () => {
-        it('returns Command Runner', async () => {
+        it('returns URL Command Runner', async () => {
             const testInput: ScanArguments = { url: 'https://www.bing.com', output: '/users/xyz' };
-            containerMock.setup((cm) => cm.get(CommandRunner)).returns(() => commandRunnerMock.object);
-            const runCommand = jest.spyOn(commandRunnerMock.object, 'runCommand').mockImplementationOnce(async () => Promise.resolve());
+            containerMock.setup((cm) => cm.get(URLCommandRunner)).returns(() => urlCommandRunnerMock.object);
+            const runCommand = jest.spyOn(urlCommandRunnerMock.object, 'runCommand').mockImplementationOnce(async () => Promise.resolve());
             // tslint:disable-next-line: no-floating-promises
             await testSubject.runScan(testInput);
             expect(runCommand).toBeCalled();
         });
+        it('returns File Command Runner', async () => {
+            const testInput: ScanArguments = { inputFile: 'inputFile.txt', output: '/users/xyz' };
+            containerMock.setup((cm) => cm.get(FileCommandRunner)).returns(() => fileCommandRunnerMock.object);
+            const runCommand = jest.spyOn(fileCommandRunnerMock.object, 'runCommand').mockImplementationOnce(async () => Promise.resolve());
+            // tslint:disable-next-line: no-floating-promises
+            await testSubject.runScan(testInput);
+            expect(runCommand).toBeCalled();
+        });
+
+        it('returns null', async () => {
+            const testInput: ScanArguments = { output: '/users/xyz' };
+            // tslint:disable-next-line: no-floating-promises
+            await testSubject.runScan(testInput).catch((error) => {
+                expect((error as Error).message).toEqual('You should provide either url or inputFile parameter only!');
```

Comment:
have you verified it will fail if the error message doesnt match "You should provide either url or inputFile parameter only!" ?

Reply:
you could do something like this https://jestjs.io/docs/en/expect#rejects

await expect(Promise.reject(new Error('octopus'))).rejects.toThrow('octopus');

Reply:
yes I did, if fails if the message doesn't match
actually the pattern you mentioned doesn't work well, it always pass if the message doesn't match